components/MapContainer/MapContainer.tsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import { Select } from '../Select';
+import { Map } from '../Map';
+import ugandaDistricts from '../../geoJSON/district.json';
+import ugandasubcounties from '../../geoJSON/subcounty.json';
+import * as distance from 'jaro-winkler';
+import * as turf from '@turf/turf';
+
+const mapContainerStyle = {
+  padding: '50px'
+};
+
+interface State {
+  leaflet: any;
+  map: any;
+  selectedDistrict: string;
+  boundaryType: string;
+  subcountyDropdownOptions: any[];
+  selectedSubcounty: string;
+}
+
+class MapContainer extends React.Component<{}, State> {
```

Comment:
Every component should be a function, with hooks where applicable.

Reply:
Updated to use hooks